src/other/broadcasting.jl

```diff
@@ -96,6 +96,14 @@ Base.maybeview(df::AbstractDataFrame, idx::CartesianIndex{2}) = df[idx]
 Base.maybeview(df::AbstractDataFrame, row::Integer, col::ColumnIndex) = df[row, col]
 Base.maybeview(df::AbstractDataFrame, rows, cols) = view(df, rows, cols)
 
+function Base.dotview(df::DataFrame, ::Colon, cols::ColumnIndex)
+    haskey(index(df), cols) && return view(df, :, cols)
+    if !(cols isa Symbol)
+        throw(ArgumentError("creating new columns using an integer index by broadcasting is disallowed"))
```

Comment:
Is  "broadcasting" really relevant here? The same problem would happen with plain assignment, right?
```suggestion
        throw(ArgumentError("creating new columns using an integer index is disallowed"))
```

Reply:
right